### ISSUE TITLE
A bug in core/conversions which makes the module fail in higher version of PyTorch

### ISSUE THREAD

**Issue by AnkaChan** [kornia/kornia#590]
## 🐛 Bug

There is a function called "rotation_matrix_to_quaternion", 
in line 302 ~ line 304, there is code like following:
```
    mask_c1 = mask_d2 * (1 - mask_d0_d1)
    mask_c2 = (1 - mask_d2) * mask_d0_nd1
    mask_c3 = (1 - mask_d2) * (1 - mask_d0_nd1)
```

The "1 - boolean tensor " operation will cause PyTorch to raise an error:
_Subtraction, the `-` operator, with a bool tensor is not supported. If you are trying to invert a mask, use the `~` or `logical_not()` operator instead._

from my understanding, the "1 -  boolean tensor" is doing the same thing as logical not. So I replaced that piece of code as:
```
    mask_c1 = mask_d2 * torch.logical_not( mask_d0_d1)
    mask_c2 = torch.logical_not( mask_d2) * mask_d0_nd1
    mask_c3 = torch.logical_not( mask_d2) * torch.logical_not( mask_d0_nd1)
```
And it worked.

## Environment

 - PyTorch Version (1.51):
 - OS (Windows):

**Comment by edgarriba** [comment]
@AnkaChan could you send a PR with this fix ? Besides, take into account that we have known issues for this set of conversion functions

**Comment by edgarriba** [comment]
@AnkaChan Any idea where the exact issue might be ?
